src/main/scala/io/treeverse/clients/LakeFSInputFormat.scala

```diff
@@ -97,26 +89,30 @@ class LakeFSInputFormat extends InputFormat[Array[Byte], WithIdentifier[Catalog.
   import LakeFSInputFormat._
 
   override def getSplits(job: JobContext): java.util.List[InputSplit] = {
-    val pathPrefix = job.getConfiguration.get(FileInputFormat.INPUT_DIR)
-    val p = new Path(pathPrefix)
+    val conf = job.getConfiguration
+    val repoName = conf.get(LAKEFS_CONF_JOB_REPO_NAME_KEY)
+    val commitID = conf.get(LAKEFS_CONF_JOB_COMMIT_ID_KEY)
+    val apiClient = new ApiClient(conf.get(LAKEFS_CONF_API_URL_KEY),
+      conf.get(LAKEFS_CONF_API_ACCESS_KEY_KEY),
+      conf.get(LAKEFS_CONF_API_SECRET_KEY_KEY))
+    val storageNamespace = apiClient.getRepositoryStorageNamespace(repoName)
+    val metaRangeURL = apiClient.getMetaRangeURL(repoName, storageNamespace, commitID)
+    val p = new Path(metaRangeURL)
     val fs = p.getFileSystem(job.getConfiguration)
     val localFile = File.createTempFile("lakefs", "metarange")
     fs.copyToLocalFile(p, new Path(localFile.getAbsolutePath))
-    // TODO(johnnyaug) ask lakeFS metadata server for the sstable url
     val rangesReader = new SSTableReader(localFile.getAbsolutePath, RangeData.newBuilder().build())
     localFile.delete()
-
     val ranges = read(rangesReader)
-
     ranges.map(
-      r => new GravelerSplit(r.message, new Path(p.getParent(), new String(r.id)))
+      r => new GravelerSplit(r.message, new Path(storageNamespace + "/" + apiClient.getRangeURL(repoName, new String(r.id))))
```

Comment:
Dunno if we should call `getRangeURL` from here or when we use the split, on a worker machine.

Advantages of doing it here:
1. If some `getRangeURL` has a hard failure (currently impossible), then we don't start a big Spark job for nothing.
2. If/when we replace the single-range query with a query that gives client hints that avoid further calls, we can use that hint in more locations.

Advantages of doing in on the workers:
1. More self contained client?
2. Better performance when we shard lakeFS servers.
3. Naturally space out calls to `getRangeURL`, because they don't all happen at once.

I think I'd rather do it on the workers, but I am more asking than requesting.

Reply:
I think let's defer this decision.